pennylane/workflow/execution.py

```diff
@@ -543,10 +545,16 @@ def cost_fn(params, x):
         )
 
     gradient_kwargs = gradient_kwargs or {}
+    mcm_config = mcm_config or {}
     config = config or _get_execution_config(
-        gradient_fn, grad_on_execution, interface, device, device_vjp
+        gradient_fn, grad_on_execution, interface, device, device_vjp, mcm_config
     )
 
+    if interface == "jax-jit" and config.mcm_config.mcm_method == "deferred":
+        # This is a current limitation of defer_measurements. "hw-like" behaviour is
+        # not yet accessible.
+        config.mcm_config.postselect_mode = "fill-shots"

```

Comment:
Should a user be warned about this? :thinking: I know it's in the docs, but a warning might prompt a confused user to look at them :)

Reply:
Yeah, me too. Personally, I wouldn't mind even raising an error, but that might inhibit usability. Discussion would be good.

Reply:
This has the potential to be annoying, always having to manually change the method from its default value.

Maybe it makes for sense for the default to be `None`? Then the "true" default can be determined smartly, like setting "hw-like" normally, unless jax is used then setting to "fill-shots" (assuming "deferred" is in effect). And if the user supplies their own value, for example by setting "hw-like" while using jax and "deferred", we can raise a clear "unsupported" error.

Reply:
I agree with @dime10 about not raising an error However, the more I think about it the more I'm leaning towards not changing anything. I have already documented that when using `mcm_method="deferred"` with shots, postselection behaviour will mimic `"fill-shots"` regardless of what `postselect_mode` is set to be (or if it is even specified at all). This discussion spawned strictly because the documented behaviour was not happening if `diff_method != None`, which has now been resolved. Let me know what everyone thinks.

Reply:
I think raising an error when the user explicitly requests hardware like shots (rather than the determination being made automatically) when it is not possible makes the most sense.

Reply:
Updated to raise error.